packages/e2e/test/wallet/PersonalWallet/handle.test.ts

```diff
@@ -0,0 +1,271 @@
+    await walletReady(wallet, coins);
+    const derivationPath = {
+      index: 2,
+      role: KeyRole.External
+    };
+    const keyAgent = await createStandaloneKeyAgent(
+      env.KEY_MANAGEMENT_PARAMS.mnemonic.split(' '),
+      await firstValueFrom(wallet.genesisParameters$),
+      await wallet.keyAgent.getBip32Ed25519()
+    );
+    const pubKey = await keyAgent.derivePublicKey(derivationPath);
+    const keyHash = await keyAgent.bip32Ed25519.getPubKeyHash(pubKey);
+    policySigner = new util.KeyAgentTransactionSigner(keyAgent, derivationPath);
+    policyScript = {
+      __type: Cardano.ScriptType.Native,
+      kind: Cardano.NativeScriptKind.RequireAllOf,
+      scripts: [
+        {
+          __type: Cardano.ScriptType.Native,
+          keyHash,
+          kind: Cardano.NativeScriptKind.RequireSignature
+        }
+      ]
+    };
+    policyId = nativeScriptPolicyId(policyScript);
+    wallet.shutdown();
+  };
+
+  const mint = async (tokens: Cardano.TokenMap, txMetadatum: Cardano.Metadatum) => {
+    const auxiliaryData = {
+      blob: new Map([[721n, txMetadatum]])
+    };
+
+    const txProps: InitializeTxProps = {
+      auxiliaryData,
+      mint: tokens,
+      outputs: new Set([
+        {
+          address: walletAddress,
+          value: {
+            assets: tokens,
+            coins
+          }
+        }
+      ]),
+      witness: { extraSigners: [policySigner], scripts: [policyScript] }
+    };
+
+    const unsignedTx = await wallet.initializeTx(txProps);
+
+    const finalizeProps: FinalizeTxProps = {
+      auxiliaryData,
+      tx: unsignedTx,
+      witness: { extraSigners: [policySigner], scripts: [policyScript] }
+    };
+
+    const signedTx = await wallet.finalizeTx(finalizeProps);
+    await submitAndConfirm(wallet, signedTx);
+  };
+
+  const restartWallet = async () => {
+    wallet.shutdown();
+    wallet = (
+      await getWallet({
+        env,
+        handlePolicyIds: [policyId],
+        idx: 0,
+        logger,
+        name: 'Minting Wallet',
+        polling: { interval: 50 }
+      })
+    ).wallet;
+    await walletReady(wallet, coins);
+  };
+
+  beforeAll(async () => {
+    await initPolicyId();
+    wallet = (
+      await getWallet({
+        env,
+        handlePolicyIds: [policyId],
+        idx: 0,
+        logger,
+        name: 'Minting Wallet',
+        polling: { interval: 50 }
+      })
+    ).wallet;
+    receivingWallet = (
+      await getWallet({
+        env,
+        handlePolicyIds: [policyId],
+        idx: 1,
+        logger,
+        name: 'Receiving Wallet',
+        polling: { interval: 50 }
+      })
+    ).wallet;
+    await Promise.all([walletReady(wallet, coins), walletReady(receivingWallet, 0n)]);
+    assetIds = [Cardano.AssetId(`${policyId}${assetNames[0]}`), Cardano.AssetId(`${policyId}${assetNames[1]}`)];
+    walletAddress = (await firstValueFrom(wallet.addresses$))[0].address;
+  });
+
+  afterEach(async () => {
+    await burnTokens({
```

Comment:
Burning the tokens after the test makes the state unusable for integration test requirements, so best to not do it in this suite.

Reply:
After discussing this on a call, it's better to separate the demands of the integration tests with the specific scenarios expressed here and just mint handles as part of setting up the initial state.

https://github.com/input-output-hk/cardano-js-sdk/tree/master/packages/e2e/local-network/scripts